### ISSUE TITLE
[Question] Have some problem for time series data clustering?

### ISSUE THREAD

**Issue by duangrux** [rtavenar/tslearn#78]
I want to do cluster analysis for 200 crypto-currency and use "tslearn" module.
each currency has 365 records but some currency has less
I like to use 3 cluster analysis methods for comparing.

my question is 
1. barycenter computing belongs to cluster analysis method? if yes, this method can cluster currency data?

2.Have any function for validation K-value?
for identifying the suitable  K-value that is the best to do cluster analysis.

3. In this module, have any function for cluster validation that show which method is the best?

4.in below the picture, Have  any thing wrong with this code, why the result is "nan"?
my data shape is (200, 365, 1) from>>> print(formatted_dataset.shape)

for the bellow code, I like to use DTW distance to measure, how i can chose DTW ?

#K shape
seed = 0
numpy.random.seed(seed)
formatted_dataset = TimeSeriesScalerMeanVariance().fit_transform(formatted_dataset[:200])  
sz = formatted_dataset.shape[1]

#Euclidean k-means
ks = KShape(n_clusters=3, verbose=True, random_state=seed)
y_pred = ks.fit_predict(formatted_dataset)

plt.figure()
for yi in range(3):
    plt.subplot(3, 1, 1 + yi)
    for xx in formatted_dataset[y_pred == yi]:
        plt.plot(xx.ravel(), "k-", alpha=.2)
    plt.plot(ks.cluster_centers_[yi].ravel(), "r-")
    plt.xlim(0, sz)
    plt.ylim(-4, 4)
    plt.title("Cluster %d" % (yi + 1))

plt.tight_layout()
plt.show()
![screen shot 2561-11-04 at 16 25 32](https://user-images.githubusercontent.com/43942541/47961926-04101080-e04f-11e8-8317-05037d39e4fb.png)
![screen shot 2561-11-04 at 16 25 39](https://user-images.githubusercontent.com/43942541/47961928-0d00e200-e04f-11e8-827e-9964a70309be.png)

@rth @pnavaro @konstantinstadler @jscheithe

**Comment by rtavenar** [comment]
Clustering with DTW as the metric can be done using TimeSeriesKMeans : https://tslearn.readthedocs.io/en/latest/auto_examples/plot_kmeans.html#sphx-glr-auto-examples-plot-kmeans-py

**Comment by duangrux** [comment]
thank you for your reply


Sent from Yahoo Mail. Get the app 

    On Tuesday, August 27, 2019, 02:18:22 AM GMT+7, Romain Tavenard <notifications@github.com> wrote:  
 
 
Clustering with DTW as the metric can be done using TimeSeriesKMeans : https://tslearn.readthedocs.io/en/latest/auto_examples/plot_kmeans.html#sphx-glr-auto-examples-plot-kmeans-py
